Session.h

```diff
@@ -70,13 +70,23 @@ class Session {
         OnSetImageChannels(request.first);
     }
     void CancelSetHistRequirements() {
-        _histogram_progress.fetch_and_store(HISTOGRAM_CANCEL);
+        _histo_context.cancel_group_execution();
+    }
+    tbb::task_group_context& HistoContext() {
```

Comment:
This seems like a weird abbreviation. I'd suggest either `Hist` (which is already used elsewhere) or writing `Histogram` out in full.